python/hidet/distributed/store.py

```diff
@@ -0,0 +1,210 @@
+            else:
+                self._cache[k] = v
+
+    def _add(self, key: str, amount: int) -> int:
+        with self._lock:
+            with open(self._filename, "ab+") as f:
+                f.seek(0)
+                self._update(f)
+            value = int(self._cache.get(key, '0')) + amount
+            with open(self._filename, "ab+") as f:
+                self._write(f, bytes(key, encoding='raw_unicode_escape'))
+                self._write(f, bytes(str(value), encoding='raw_unicode_escape'))
+        return value
+
+    def _check(self, keys: List[str]):
+        with self._lock:
+            with open(self._filename, "ab+") as f:
+                f.seek(0)
+                self._update(f)
+        return all((key in self._cache for key in keys))
+
+    def _set(self, key: str, value: bytes):
+        with self._lock:
+            with open(self._filename, "ab+") as f:
+                self._write(f, bytes(key, encoding='raw_unicode_escape'))
+                self._write(f, value)
+
+    def set(self, key: str, value: bytes) -> None:
+        self._set(self.REGULAR_PREFIX + key, value)
+
+    def get(self, key: str) -> bytes:
+        last_file_size = None
+        key = self.REGULAR_PREFIX + key
+        start_t = datetime.now()
+        while True:
+            self._lock.acquire()
+            with open(self._filename, "ab+") as f:
+                f.seek(0)
+                file_size = self._file_size(f)
+                if key not in self._cache and file_size == last_file_size:
+                    # No new entries
+                    last_file_size = file_size
+                    self._lock.release()
+                    if self._timeout is not None and datetime.now() - start_t > self._timeout:
+                        raise TimeoutError()
+                    time.sleep(0.01)
+                    continue
+                last_file_size = file_size
+                self._update(f)
+                self._lock.release()
+                value = self._cache.get(key)
+                if value is not None:
+                    return value
+
+    def add(self, key: str, amount: int) -> int:
+        return self._add(self.REGULAR_PREFIX + key, amount)
+
+    def compare_set(self, key: str, expected: bytes, desired: bytes) -> bytes:
+        key = self.REGULAR_PREFIX + key
+        with self._lock:
+            with open(self._filename, "ab+") as f:
+                f.seek(0)
+                self._update(f)
+                has_key = key in self._cache
+                print(has_key, self._cache[key])
+                if (not has_key and expected == b'') or (has_key and self._cache[key] == expected):
+                    f.seek(0, 2)
+                    self._write(f, bytes(key, encoding='raw_unicode_escape'))
+                    self._write(f, desired)
+                    return desired
+                elif not has_key:
+                    return expected
+        return self._cache[key]
```

Comment:
```suggestion
                else:
                    return self._cache[key]
```

Reply:
Seems the pytorch documentation does not mention the return value of this function. Consider adding some comments for the return value (if we are different from pytorch's behaviour).

Reply:
Thanks for the clarification!